### ISSUE TITLE
Slow.....

### ISSUE THREAD

**Issue by zam6ak** [bjrmatos/jsreport-electron-pdf#6]
I have a report which uses plain HTML table and handlebars.
Report has 8000+ rows. No scripts of any kind - just pure HTML to PDF.

With phantomjs it takes about 15 seconds to render.
I tried using electron recipe with 'electron-ipc' and 'electron-server' strategy and
the same report takes 40+ seconds to render....

Any idea why would it be that much slower?

**Comment by bjrmatos** [comment]
@zam6ak hmmm i have experienced slow rendering on windows machines, can you upload a working demo on a repository for me? it will help me to figure out why it is slow.

**Comment by zam6ak** [comment]
Unfortunately I can't share the data....
You can try to generate fake data using https://www.mockaroo.com/
As for the template - it is just handlebars with HTML table....

**Comment by pofider** [comment]
Yes electron seems to be really slower (maybe only in some cases), but I'm not sure if there is something we can do about it. The slowdown likely comes from deep down somewhere in chromium codebase.

@bjrmatos  if you have any clue what to improve, you can compare the speed on this simple example
https://playground.jsreport.net/studio/workspace/ryQffOb3/4

**Comment by bjrmatos** [comment]
@pofider thnk you for the example, it will help me to figure out!

**Comment by bjrmatos** [comment]
interesting, using https://playground.jsreport.net/studio/workspace/ryQffOb3/4 my numbers on OSX are:

phantomjs (using `phantom-server`strategy)
114 pages
first render -> 29941 seconds
second render -> 28805 seconds

electron (using `electron-server`strategy)
107 pages 
first render -> 11489 seconds
second render -> 11208 seconds

phantomjs (using `dedicated-process`strategy)
114 pages
1rst -> 30198 seconds
2rst -> 29898 seconds

electron (using `dedicated-process`strategy)
107 pages 
1rst -> 11600 seconds
2rst -> 11344 seconds

i need to get a windows machine to investigate more, but yeah it is slow on windows 😆

**Comment by notatestuser** [comment]
I can't get this package to work at all on Windows. I always end up with a `Error: Timeout when executing in electron` regardless of the strategy I choose. Do you know why this might be?

**Comment by zam6ak** [comment]
did you do:
`npm install electron-prebuilt --save`

**Comment by notatestuser** [comment]
I did, yes.

**Comment by notatestuser** [comment]
This is when running under `iisnode`. When ran directly it works! I'll consider this a file access permission issue.

**Comment by pofider** [comment]
You can try to change this in the config :
```js
electron: { 
  strategy: 'dedicated-process',
}
```

However I would recommend avoid using `iisnode` and rather use `iis` as reverse proxy to a windows service running jsreport. If you are not limited by infrastructure rules of course..

**Comment by bjrmatos** [comment]
@notatestuser we ( @pofider and I ) have tested `jsreport-electron-pdf` on windows with iisnode and it causes too much troubles, seems like electron on iisnode doesn't work well :( you can try with @pofider 's recomendations.

**Comment by notatestuser** [comment]
Indeed. I've tried all the strategies without any luck. Interestingly though phantomjs is fine, even with a prebuilt binary downloaded into node_modules. I'd really like to be able to render in electron!

**Comment by notatestuser** [comment]
Have tried to use `execFile` instead of `spawn` in server mode - still no joy.

**Comment by notatestuser** [comment]
I've opened bjrmatos/electron-workers#263 for the iisnode issue.
